src/netmessagemaker.h

```diff
@@ -12,14 +12,14 @@
 class CNetMsgMaker
 {
 public:
-    explicit CNetMsgMaker(int nVersionIn) : nVersion(nVersionIn){}
+    explicit CNetMsgMaker(int /*unused*/) {}
```

Comment:
There's not really any need for a `NetMsgMaker` objects anymore after this change -- `NetMsgMaker::Make()` could just be a global/static function.

Reply:
Fair enough

Reply:
Could move `NetMsgMaker` entirely into `PushMessage` at that point, perhaps. *shrug*

Reply:
> Could move `NetMsgMaker` entirely into `PushMessage` at that point, perhaps. _shrug_

FWIW, NetMsgMaker was originally created exactly to prevent this for the sake of logical separation. The idea is/was that `net` shouldn't have any idea how to serialize a `CBlock`, rather it should just receive the dumb bytes.

Reply:
@theuni I think my idea there was that `template<typename T...> PushMessage(CNode& node, T... stuff)` that accepts `CBlock` and so on would be in net_processing.cpp, and that would call `CNode::PushMessage(msg)` after serializing? That would fit with https://github.com/ajtowns/bitcoin/commit/6be38ef045cd1b974ae13d705260e83b769a48b5 at least. I'm just making this up as I go along though :man_shrugging: